app/Foliage/PrepareSource.hs

```diff
@@ -117,16 +117,32 @@ addPrepareSourceRule inputDir cacheDir = addBuiltinRule noLint noIdentity run
           applyMSubdir = case mSubdir of Just s -> (</> s); _ -> id
           srcDir = applyMSubdir $ byPassSingleTopLevelDir tmpDir
 
-      cmd_
-        [ "cp"
-        , -- copy directories recursively
-          "--recursive"
-        , -- treat DEST as a normal file
-          "--no-target-directory"
-        , -- always follow symbolic links in SOURCE
-          "--dereference"
-        , -- SOURCE
-          srcDir
-        , -- DEST
-          outDir
-        ]
+      copyDirectoryContents srcDir outDir
+
+gitCheckout :: FilePath -> GitHubRepo -> GitHubRev -> Action FilePath
+gitCheckout cacheDir repo rev = do
+  alreadyCloned <- doesDirectoryExist path
+  if alreadyCloned
+    then command_ [Cwd path] "git" ["fetch"]
+    else command_ [] "git" ["clone", "--recursive", url, path]
+  command_ [Cwd path] "git" ["checkout", show rev]
+  command_ [Cwd path] "git" ["submodule", "update"]
```

Comment:
AFAIU this relies on the submodules already having been initialised right? Otherwise we need to pass `--init` or add `git submodule init` before it.

Reply:
A `--recursive` clone does initialize submodules.